Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -547,6 +547,19 @@
         return;
     }
 
+    // Other than smart replies, all Writing Tools call this method twice for the last replacement,
+    // varying only the finished parameter. In this case, simply commit the overall replacement
+    // to the undo stack, rather than performing a redundant replacement.
+    if (finished && session.compositionType != WritingTools::SessionCompositionType::SmartReply) {
+        if (RefPtr document = this->document()) {
```

Comment:
I think this document check should be moved to the top of the function, outside of any outer if statement, because if there isn't a document we should just early return regardless